From fd61e53ed08cec2b3af688a55507477f6ef30544 Mon Sep 17 00:00:00 2001
From: Cyril Brulebois <kibi@debian.org>
Date: Fri, 3 Jun 2011 15:59:14 +0200
Subject: =?UTF-8?q?Avoid=20=E2=80=9C'ReverseConvertProc'=20defined=20but?=
 =?UTF-8?q?=20not=20used=E2=80=9D=20warning.?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Move this function next to its unique caller (AceCadPreInit), and only
declare/define it when the input ABI is strictly less than 12.

Reviewed-by: Peter Hutterer <peter.hutterer@who-t.net>
Reviewed-by: Daniel Stone <daniel@fooishbar.org>
Signed-off-by: Cyril Brulebois <kibi@debian.org>

diff --git a/src/acecad.c b/src/acecad.c
index 6259f21..f5a2213 100644
--- src/acecad.c
+++ src/acecad.c
@@ -370,6 +370,21 @@ SetupProc_fail:
     return NULL;
 }
 
+static Bool
+ReverseConvertProc (InputInfoPtr local,
+        int x, int  y,
+        int *valuators)
+{
+    AceCadPrivatePtr priv = (AceCadPrivatePtr)(local->private);
+
+    // xf86Msg(X_INFO, "%s: reverse coordinate conversion in : %d, %d\n", local->name, x, y);
+    valuators[0] = x * priv->acecadMaxX / screenInfo.screens[0]->width;
+    valuators[1] = y * priv->acecadMaxY / screenInfo.screens[0]->height;
+    // xf86Msg(X_INFO, "%s: reverse coordinate conversion out: %d, %d\n", local->name, valuators[0], valuators[1]);
+
+    return TRUE;
+}
+
 static int
 NewAceCadPreInit(InputDriverPtr drv, InputInfoPtr local, int flags)
 #else
@@ -1071,22 +1086,6 @@ ConvertProc (InputInfoPtr local, int first, int num,
 }
 
 
-static Bool
-ReverseConvertProc (InputInfoPtr local,
-        int x, int  y,
-        int *valuators)
-{
-    AceCadPrivatePtr priv = (AceCadPrivatePtr)(local->private);
-
-    // xf86Msg(X_INFO, "%s: reverse coordinate conversion in : %d, %d\n", local->name, x, y);
-    valuators[0] = x * priv->acecadMaxX / screenInfo.screens[0]->width;
-    valuators[1] = y * priv->acecadMaxY / screenInfo.screens[0]->height;
-    // xf86Msg(X_INFO, "%s: reverse coordinate conversion out: %d, %d\n", local->name, valuators[0], valuators[1]);
-
-    return TRUE;
-}
-
-
 #define WriteString(str)\
     XisbWrite (priv->buffer, (unsigned char *)(str), strlen(str))
 
diff --git a/src/acecad.h b/src/acecad.h
index a2b5c66..bd8e997 100644
--- src/acecad.h
+++ src/acecad.h
@@ -102,12 +102,12 @@ static Bool DeviceClose (DeviceIntPtr);
 static Bool DeviceInit (DeviceIntPtr);
 static void ReadInput (InputInfoPtr);
 static Bool ConvertProc (InputInfoPtr, int, int, int, int, int, int, int, int, int *, int *);
-static Bool ReverseConvertProc(InputInfoPtr , int , int , int*);
 static Bool QueryHardware (AceCadPrivatePtr);
 static void NewPacket (AceCadPrivatePtr priv);
 static Bool AceCadGetPacket (AceCadPrivatePtr);
 #if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 12
 static InputInfoPtr AceCadPreInit(InputDriverPtr, IDevPtr , int);
+static Bool ReverseConvertProc(InputInfoPtr , int , int , int*);
 #else
 static int AceCadPreInit(InputDriverPtr, InputInfoPtr , int);
 #endif
-- 
cgit v0.10.2

